lib/init.coffee

```diff
@@ -46,20 +56,19 @@ module.exports =
             '--quiet',
             tmpFilePath
           ]
-          return helpers.exec(@executablePath, params, {cwd}).then (stdout) ->
-            regex = /Line\s(\d+), E:([^:]+):\s(.+)/
-            lines = stdout.split('\n').filter (line) ->
-              line.indexOf('Line') is 0
+          return helpers.exec(@executablePath, params, {cwd}).then (stdout) =>
+            regex = XRegExp '^(?<line>\\d+), E:(?<code>[^:]+): (?<message>.+)$',
+              's'
+
+            lines = stdout.split(/\nLine\s/)
 
             return [] unless lines.length
-            return lines.map (msg) ->
-              res = regex.exec(msg)
-              [all, line, code, text] = res
-              line = parseInt(line, 10)
-              text = 'E:' + code + ' - ' + text
-              return {
-                text: text,
-                type: 'error',
-                filePath: filePath,
-                range: helpers.rangeFromLineNumber(editor, line - 1)
-              }
+
+            messages = []
+
+            lines.forEach (msg) =>
+              XRegExp.forEach msg, regex, (match, i) =>
```

Comment:
I assumed so, but need to doublecheck - good catch!

On Mon, Oct 19, 2015, 15:24 Steel Brain <notifications@github.com> wrote:

> In lib/init.coffee
> <https://github.com/AtomLinter/linter-gjslint/pull/21#discussion_r42376696>
> :
>
> > -            return lines.map (msg) ->
> > -              res = regex.exec(msg)
> > -              [all, line, code, text] = res
> > -              line = parseInt(line, 10)
> > -              text = 'E:' + code + ' - ' + text
> > -              return {
> > -                text: text,
> > -                type: 'error',
> > -                filePath: filePath,
> > -                range: helpers.rangeFromLineNumber(editor, line - 1)
> > -              }
> > +
> > +            messages = []
> > +
> > +            lines.forEach (msg) =>
> > +              XRegExp.forEach msg, regex, (match, i) =>
>
> Do you ever get more than one error per line? If not, then you don't need
> this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AtomLinter/linter-gjslint/pull/21/files#r42376696>.
>